src/cmd/services/m3coordinator/downsample/metrics_appender.go

```diff
@@ -103,8 +120,13 @@ func newMetricsAppender(pool *metricsAppenderPool) *metricsAppender {
 // reset is called when pulled from the pool.
 func (a *metricsAppender) reset(opts metricsAppenderOptions) {
 	a.metricsAppenderOptions = opts
-	if a.tagEncoder == nil {
-		a.tagEncoder = opts.tagEncoderPool.Get()
+
+	// Copy over any previous inuse encoders to the cached encoders list. If
+	// there are no cached encoders then create one.
+	a.cachedEncoders = append(a.cachedEncoders, a.inuseEncoders...)
+	a.inuseEncoders = a.inuseEncoders[:0]
+	if len(a.cachedEncoders) == 0 {
+		a.cachedEncoders = append(a.cachedEncoders, opts.tagEncoderPool.Get())
```

Comment:
nit: Do you need to do this? Why not let "one place" do this perhaps, in the call to `a.tagEncoder()` which already lazy allocs the tag encoder?

Reply:
true, dont need this here. will remove.